docs/cookbook/create-basic-grid.md

```diff
@@ -0,0 +1,270 @@
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
+        ...
+
++       $grid->setTheme('@ShopsysShop/Admin/Content/Dashboard/articleGrid.html.twig');
+
+        return $grid;
+    }
+```
+
+## Sort rows
+Now we see data as we want, but it would be nice to be able to adjust the view from the users perspective.
+We may want to ease finding of a certain article by allowing to sort rows by name or date of creation.
+
+A default order will be by date with newest articles at the top.
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
+        $grid->addColumn('id', 'a.id', t('Id'));
+-       $grid->addColumn('name', 'a.name', t('Title'));
++       $grid->addColumn('name', 'a.name', t('Title'), true);
+-       $grid->addColumn('createdAt', 'a.createdAt', t('Created'));
++       $grid->addColumn('createdAt', 'a.createdAt', t('Created'), true);
+
++       $grid->setDefaultOrder('createdAt', DataSourceInterface::ORDER_DESC);
+    ...
+```
+
+## Paginate results
+As the number of articles grow, the clarity decreases rapidly.
+To keep work with grid enjoyable, we can split data across several pages with pagination.
+
+In the grid is just a matter of calling one method.
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
++       $grid->enablePaging();
+
+        ...
+```
+
+## Delete articles from grid
+As the last, we may want to delete some articles.
+Grid eases the task with already implemented action column.
+
+We just use `addDeleteActionColumn` in existing `ArticleGridFactory` with arguments `admin_dashboard_delete` as a route (to handle request – we create proper action in a moment) and request parameters (action have to know id of the article to delete).
+To prevent accidental deletion we can also set a confirmation message.
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
+        $grid->addColumn('id', 'a.id', t('Id'));
+        $grid->addColumn('name', 'a.name', t('Title'));
+        $grid->addColumn('createdAt', 'a.createdAt', t('Created'));
+
++       $grid->addDeleteActionColumn('admin_dashboard_delete', ['id' => 'a.id'])
++           ->setConfirmMessage(t('Really delete?'));
+    ...
+```
+
+And now we add the action to delete the article.
+
+```php
+...
+use Shopsys\FrameworkBundle\Component\Router\Security\Annotation\CsrfProtection;
+
+class DashboardController extends AdminBaseController
+{
+    ...
+
+    /**
+     * @Route("/dashboard/articles/delete/{id}", requirements={"id" = "\d+"})
+     * @CsrfProtection
+     * @param int $id
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function deleteAction(int $id): Response
+    {
+        // delete article
```

Comment:
this won't be implemented here? It is really easy as there is a method prepared for that in `ArticleFacade` :grin: